### DISCUSSION TITLE
How to add an image to a post folder?

### DISCUSSION THREAD

**Discussion by SeiOkami** [cotes2020/jekyll-theme-chirpy#1675] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

I put the image in my post folder. What is its url adress?
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/42138875/b6da8f5d-7aaf-43b3-9af7-8936fca097f4)

### Operations you have already tried

/img/01.png

**Comment by steveng57** [comment]
Usually, your images and any other static content does NOT go in the /_posts folder, but rather it goes, in the /assets folder.  For images, this is usually under /assets/img  

Then you can include the images in your post by referencing them via /assets/img/01.png   

Of course, you can create any directory/folder structure you want under /assets/img and if you have any complexity in your site or any substantial number of images, you should probably do just that.  I have 4 sub folders under /assets/img and dozens more under those folders as well.

**Reply by LucienShui** [reply]
What if I want to preview the markdown render result locally? The `/assets/img` would misleading the local markdown editor but correct for browser.

**Comment by kungfux** [comment]
I may be wrong, but images from the `_posts` folder will not be processed by default. If you really want to use this folder for static files, you have to change your configuration according to the [Jekyll documentation](https://jekyllrb.com/docs/static-files/). However, since you are using a subfolders structure there, this may not be possible. You'll have to find out if this is possible without using some plugins or writing some code. Will leave a few extra links here: [one](https://jekyllrb.com/docs/structure/), [two](https://github.com/jekyll/jekyll/discussions/8692), [three](https://stackoverflow.com/questions/74039848/include-images-in-jekyll-collection).

If you're just not sure where to place image files in general, please follow defaults according to @steveng57 suggestion above.

**Comment by SeiOkami** [comment]
Do I understand correctly that according to Jekyll logic it is necessary to place files like this?

- Store text of posts in _posts/PostName
- Store post images in /assets/img/PostName/

**Reply by lotusk08** [reply]
Yes, you can try it.
